feat(amplify): support custom certificate (#30791)
aws/aws-cdk: packages/@aws-cdk/aws-amplify-alpha/README.md

domain.mapSubDomain(dev); // sub domain prefix defaults to branch name
```

> If you want to use your custom domain, set your custom certificate to `customCertificate`:

Reviewer: ```suggestion
To specify a custom certificate for your custom domain use the `customCertificate` property:
```
